packaging/grafana-agent-flow/linux_packages_test.go

```diff
@@ -0,0 +1,267 @@
+
+	res = env.Uninstall()
+	require.Equal(t, 0, res.ExitCode, "uninstalling failed")
+
+	res = env.ExecScript(`[ -f /usr/bin/grafana-agent-flow ]`)
+	require.Equal(t, 1, res.ExitCode, "expected grafana-agent-flow to be uninstalled")
+	// NOTE(rfratto): we don't check for what happens to the config file here,
+	// sicne the behavior is inconsistent: rpm uninstalls it, but deb doesn't.
+}
+
+func EnvironmentTestConfigPersistence(t *testing.T, env Environment) {
+	res := env.ExecScript(`echo -n "keepalive" > /etc/grafana-agent.river`)
+	require.Equal(t, 0, res.ExitCode, "failed to write config file")
+
+	res = env.Install()
+	require.Equal(t, 0, res.ExitCode, "installation failed")
+
+	res = env.ExecScript(`cat /etc/grafana-agent.river`)
+	require.Equal(t, "keepalive", res.Stdout, "Expected existing file to not be overridden")
+}
+
+func EnvironmentTestDataFolderPermissions(t *testing.T, env Environment) {
+	// Installing should create /var/lib/grafana-agent-flow, assign it to the
+	// grafana-agent-flow user and grafana-agent group, and set its permissions
+	// to 0770.
+	res := env.Install()
+	require.Equal(t, 0, res.ExitCode, "installation failed")
+
+	res = env.ExecScript(`[ -d /var/lib/grafana-agent-flow ]`)
+	require.Equal(t, 0, res.ExitCode, "Expected /var/lib/grafana-agent to have been created during install")
+
+	res = env.ExecScript(`stat -c '%a:%U:%G' /var/lib/grafana-agent-flow`)
+	require.Equal(t, "770:grafana-agent-flow:grafana-agent\n", res.Stdout, "wrong permissions for data folder")
+	require.Equal(t, 0, res.ExitCode, "stat'ing data folder failed")
+}
+
+type Environment struct {
+	Install    func() ExecResult
+	Uninstall  func() ExecResult
+	ExecScript func(string) ExecResult
+}
+
+type ExecResult struct {
+	Stdout, Stderr string
+	ExitCode       int
+}
+
+// RPMEnvironment creates an Environment to install the agent RPM against.
+func RPMEnvironment(t *testing.T, pool *dockertest.Pool) Environment {
+	t.Helper()
+
+	container := environmentContainer(
+		t,
+		pool,
+		"../testdata/centos-systemd.Dockerfile",
+		"agent-test-centos-systemd",
+		fmt.Sprintf("../../dist/grafana-agent-flow-0.0.0-1.%s.rpm", runtime.GOARCH),
+	)
+
+	return Environment{
+		Install: func() ExecResult {
+			filename := fmt.Sprintf("/tmp/grafana-agent-flow-0.0.0-1.%s.rpm", runtime.GOARCH)
+			return containerExec(t, container, "rpm", "-i", filename)
+		},
+		Uninstall: func() ExecResult {
+			return containerExec(t, container, "rpm", "-e", "grafana-agent-flow")
+		},
+		ExecScript: func(script string) ExecResult {
+			return containerExec(t, container, "/bin/bash", "-c", script)
+		},
+	}
+}
+
+// DEBEnvironment creates an Environment to install the agent RPM against.
+func DEBEnvironment(t *testing.T, pool *dockertest.Pool) Environment {
+	t.Helper()
+
+	container := environmentContainer(
+		t,
+		pool,
+		"../testdata/debian-systemd.Dockerfile",
+		"agent-test-debian-systemd",
+		fmt.Sprintf("../../dist/grafana-agent-flow-0.0.0-1.%s.deb", runtime.GOARCH),
```

Comment:
Same here - we are assuming that the current working directory is the one of the _test.go file?